spec/runtime/with_unbundled_env_spec.rb

```diff
@@ -188,14 +179,14 @@ def build_bundler_context
     end
   end
 
-  describe "Bundler.clean_system", :bundler => "< 2" do
+  describe "Bundler.clean_system", :bundler => "< 3" do
     it "runs system inside with_clean_env" do
       Bundler.clean_system(%(echo 'if [ "$BUNDLE_PATH" = "" ]; then exit 42; else exit 1; fi' | /bin/sh))
       expect($?.exitstatus).to eq(42)
     end
   end
 
-  describe "Bundler.clean_exec", :bundler => "< 2" do
+  describe "Bundler.clean_exec", :bundler => "< 3" do
```

Comment:
`Bundler.clean_exec` should be tested on bundler 3.

Reply:
This uses `with_clean_env` which has been marked as been removed in Bundler 3

Reply:
You're right again.

We need a better deprecation path here, and alternatives. At the moment users using `clean_exec` will get a deprecation message about `with_clean_env`, which they are not directly using. I'll work on this on a separate PR, but this upper limit is fine.

Sorry for the bad comments on this review.